src/Datadog.Trace.ClrProfiler.Native/cor_profiler_base.cpp

```diff
@@ -14,58 +14,72 @@ CorProfilerBase::~CorProfilerBase() {
 
 HRESULT STDMETHODCALLTYPE
 CorProfilerBase::Initialize(IUnknown *pICorProfilerInfoUnk) {
+  Debug("Initialize");
   return S_OK;
 }
 
-HRESULT STDMETHODCALLTYPE CorProfilerBase::Shutdown() { return S_OK; }
+HRESULT STDMETHODCALLTYPE CorProfilerBase::Shutdown() {
+  Debug("Shutdown");
+  return S_OK;
+}
 
 HRESULT STDMETHODCALLTYPE
 CorProfilerBase::AppDomainCreationStarted(AppDomainID appDomainId) {
+  Debug("AppDomainCreationStarted: ", appDomainId);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE CorProfilerBase::AppDomainCreationFinished(
     AppDomainID appDomainId, HRESULT hrStatus) {
+  Debug("AppDomainCreationFinished: ", appDomainId, " hrStatus=", hrStatus);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE
 CorProfilerBase::AppDomainShutdownStarted(AppDomainID appDomainId) {
+  Debug("AppDomainShutdownStarted: ", appDomainId);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE CorProfilerBase::AppDomainShutdownFinished(
     AppDomainID appDomainId, HRESULT hrStatus) {
+  Debug("AppDomainShutdownFinished: ", appDomainId, " ", hrStatus);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE
 CorProfilerBase::AssemblyLoadStarted(AssemblyID assemblyId) {
+  Debug("AssemblyLoadStarted: ", assemblyId);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE
 CorProfilerBase::AssemblyLoadFinished(AssemblyID assemblyId, HRESULT hrStatus) {
+  Debug("AssemblyLoadFinished: ", assemblyId, " ", hrStatus);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE
 CorProfilerBase::AssemblyUnloadStarted(AssemblyID assemblyId) {
+  Debug("AssemblyUnloadStarted: ", assemblyId);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE CorProfilerBase::AssemblyUnloadFinished(
     AssemblyID assemblyId, HRESULT hrStatus) {
+  Debug("AssemblyUnloadFinished: ", assemblyId, " ", hrStatus);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE
 CorProfilerBase::ModuleLoadStarted(ModuleID moduleId) {
+  Debug("ModuleLoadStarted: ", moduleId);
   return S_OK;
 }
 
 HRESULT STDMETHODCALLTYPE
 CorProfilerBase::ModuleLoadFinished(ModuleID moduleId, HRESULT hrStatus) {
+  Debug("ModuleLoadFinished: ", moduleId, " ", hrStatus);
   return S_OK;
 }
 
```

Comment:
I don't know if ModuleUnloadStarted is ever called, but we could use a Debug() call there for visibility.

Reply:
`ModuleUnloadStarted()` is implemented in `cor_profiler.cpp` and has `Debug()` calls. The ones here in `cor_profiler_base.cpp` are generally the ones that I wanted to log but are not overridden there.